packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -80,6 +80,8 @@ class PerformanceMonitoring {
           var spanName =
             task.data.method + ' ' + stripQueryStringFromUrl(task.data.url)
           var span = transactionService.startSpan(spanName, 'external.http')
+          var taskId = transactionService.addTask()
```

Comment:
does this mean users will have to create a task every time they add a span?

Reply:
if yes, how about handling that behind the scenes for users? If you _really_ want to end the transaction, you should end all the spans?

Reply:
No the spans would work as they are today. i.e we don't wait for the spans to finish and if the transaction is finished we truncate the running spans.  

The users would have to add tasks for any async operation they want to wait for before ending the transaction (whether there's a span for it or not). The most common async operation are XHRs which the agent automatically adds the task for them. Going forward we will also add automatic tasks for async operations we think are useful.

We did discuss adding a parameter to the `startSpan` method to also create a task for it, but I think we should do it as future improvements.

Reply:
OK. In which situations would you like to add a span, but want to finish the transaction before the span ends? And could you, in those situations, just call `transaction.end()` instead? 

I also went looking for the motivation to add the tasks API, but didn't see anything in the original issue or here (as a side node, the description for this PR is "Add e2e tests for react app"). This PR also doesn't referer to the original issue, so people have to go hunting to find it. 

Is the problem that people who create custom spans have no way to control the transaction duration today because the TaskAPI isn't public? In that case, it seems more appropriate to add a task whenever users add a span.

Reply:
Thanks @roncohen for pointing that out! I've update the issue description and linked to it from the PR!

Please make comment on the original issue if something needs clarifying!